agent/repowise.py

```
@@ -92,30 +129,103 @@ def get_git_health(self) -> Dict[str, Any]:
+        # Look for common patterns
+        pattern_indicators = {
+            "FastAPI/REST": ["FastAPI", "router", "endpoint"],
+            "React/Frontend": ["react", "component", "useState"],
+            "Docker": ["Dockerfile", "docker-compose"],
+            "Agentic": ["agent", "loop", "tool", "prompt"]
+        }
+
+        for name, keywords in pattern_indicators.items():
+            for kw in keywords:
+                cmd = f"grep -ri '{kw}' . --include='*.py' --include='*.js' --include='*.ts' --include='Dockerfile' --include='*.yaml' --exclude-dir={{.git,__pycache__,.venv,node_modules}} | head -n 1"
+                result = subprocess.run(cmd, shell=True, cwd=self.root, capture_output=True, text=True)
```

Ruff (0.15.12)

[error] `subprocess` call with `shell=True` identified, security issue

(S602)